DeepSpeech.py

```diff
@@ -694,594 +651,999 @@ def stopwatch(start_duration=0):
+            self._epoch = step / steps_per_epoch
+
+            # Number of additional 'jobs' trained already 'on top of' our start epoch
+            jobs_trained = (step % steps_per_epoch) * batches_per_step / batches_per_job
+
+            # Total number of train/dev/test jobs covering their respective whole sets (one epoch)
+            self._num_jobs_train = max(1, data_sets.train.total_batches / batches_per_job)
+            self._num_jobs_dev   = max(1, data_sets.dev.total_batches   / batches_per_job)
+            self._num_jobs_test  = max(1, data_sets.test.total_batches  / batches_per_job)
+
+            if FLAGS.epoch < 0:
+                # A negative epoch means to add its absolute number to the epochs already computed
+                self._target_epoch = self._epoch + abs(FLAGS.epoch)
+            else:
+                self._target_epoch = FLAGS.epoch
+
+            # State variables
+            # We only have to train, if we are told so and are not at the target epoch yet
+            self._train = FLAGS.train and self._target_epoch > self._epoch
+            self._test = FLAGS.test
+
+            if self._train:
+                self._train = True
+                # The total number of jobs for all additional epochs to be trained
+                # Will be decremented for each job that is produced/put into state 'open'
+                self._num_jobs_train_left = (self._target_epoch - self._epoch) * self._num_jobs_train - jobs_trained
+                log_info('STARTING Optimization')
+                self._training_time = stopwatch()
+
+            # Important for debugging
+            log_debug('step: %d' % step)
+            log_debug('epoch: %d' % self._epoch)
+            log_debug('target epoch: %d' % self._target_epoch)
+            log_debug('steps per epoch: %d' % steps_per_epoch)
+            log_debug('batches per job: %d' % batches_per_job)
+            log_debug('batches per step: %d' % batches_per_step)
+            log_debug('number of jobs in train set: %d' % self._num_jobs_train)
+            log_debug('number of jobs already trained in first epoch: %d' % jobs_trained)
+
+            self._next_epoch()
+
+    def _next_epoch(self):
+        # State-machine of the coodination process
+
+        # Indicates, if there were 'new' epoch(s) provided
+        result = False
+
+        if self._train:
+            # We are in train mode
+            if self._num_jobs_train_left > 0:
+                # There are still jobs left
+                num_jobs_train = min(self._num_jobs_train_left, self._num_jobs_train)
+                self._num_jobs_train_left -= num_jobs_train
+
+                # Let's try our best to keep the notion of curriculum learning
+                self._reset_counters()
+
+                # If the training part of the current epoch should generate a WER report
+                is_display_step = FLAGS.display_step > 0 and (self._epoch % FLAGS.display_step == 0 or self._epoch == self._target_epoch)
```

Comment:
With this flag being calculated as is, we'll always calculate a report on epoch 0. The only time we wanna do that is if display_step is 1, so I think this line needs to be:

```python
    is_display_step = FLAGS.display_step > 0 and (FLAGS.display_step == 1 or self._epoch > 0) and (self._epoch % FLAGS.display_step == 0 or self._epoch == self._target_epoch)
```